docs/keywords.md

```diff
@@ -0,0 +1,51 @@
+  # NL-Augmenter Keywords
+
+Every transformation should have keywords that identify the type of transformation that is being written. The following list can be used as reference.
+- the type of linguistic change that the transformation is attempting eg. morphological, lexical, syntactic, word-order, discourse, noise etc.
+- the type of algorithm that is written eg. rule-based, model-based, api-based, external-knowledge-based, transformer-based, parser-based, etc.
+- the naturalness of the generation eg. natural-sounding, natural-looking, unnatural, etc.
+- the potential accuracy & precision of the generation eg. accurate, near-accurate, high-precision, high-coverage, etc.
+- the text-specific modality being covered (if applicable) eg. visual, aural, written, etc.
+- the skills required to differentiate between the input and the output of the transformation (if applicable) eg. causal-reasoning, visual-reasoning, social-reasoning, humor, figurative-language, grammar, world-knowledge, etc. 
+- other transformation specific keywords
```

Comment:
Would it make sense to have another category of keywords to specify the preprocessing prerequisites required? (or maybe we can roll those into ``type of algorithm" category. e.g POS tagger, dependency parser, chunker, EDU segmenter, RST parser, tokenizer etc etc - after all preprocessing is also part of the model - but nonetheless we should maybe have these keywords available on there from the start)

Reply:
I think Zhenhao meant high-generations for a large number of generations (i.e. outputs) while high coverage was added for how much input space is covered. I will try to make that distinction prominent.

Reply:
Okay, I had added `parser-based` as a general category but now I've added some more fine-grained ones - `tokenizer`, `chunker`, `discourse-segmenter`, etc.

Reply:
Thanks!!